backend/typescript/services/implementations/volunteerService.ts

```diff
@@ -148,6 +166,29 @@ class VolunteerService implements IVolunteerService {
     return userVolunteerDTOs;
   }
 
+  async getCheckInsAndSchedules(
+    volunteerId: string,
+  ): Promise<(CheckInDTO | SchedulingDTO)[]> {
+    const checkIns = await (
+      await this.checkInService.getCheckInsByVolunteerId(volunteerId)
+    ).map((checkIn) => ({ ...checkIn, type: ShiftType.CHECKIN }));
+    const schedulings = await (
+      await this.schedulingService.getSchedulingsByVolunteerId(volunteerId)
+    ).map((scheduling) => ({ ...scheduling, type: ShiftType.SCHEDULING }));
+    const shifts = [...checkIns, ...schedulings].sort((a, b) => {
+      const date1 =
+        "startDate" in a
```

Comment:
is this to check if the object is a checkin or scheduling? If so, can we not check the type somehow (might not be the best way), or can maybe check with the type field you added?

Reply:
i think this would be the easiest. to check the instance you would have to make `typeguards` which is more complicated than i think is needed and it checks whether or not a method exists.

in case the method changes i change it to the type i created